src/test/unittest/devdax.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright 2019, Intel Corporation
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#
+#     * Neither the name of the copyright holder nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+"""Device dax context classes and utilities"""
+
+import os
+
+import context as ctx
+import futils
+
+
+class DevDax():
+    def __init__(self, name):
+        self.name = name
+        self.path = None
+
+    def __str__(self):
+        return self.name
+
+
+class DevDaxes():
+    def __init__(self, *dax_devices):
+        self.dax_devices = tuple(dax_devices)
+        for dd in dax_devices:
+            setattr(self, dd.name, dd)
+
+    def setup(self, **kwargs):
+        tools = kwargs['tools']
+        for dd in self.dax_devices:
+            proc = tools.pmemdetect('-d', dd.path)
+            if proc.returncode != 0:
+                raise futils.Fail('checking {} with pmemdetect failed:{}{}'
+                                  .format(dd.path, os.linesep, proc.stdout))
+
+            proc = tools.pmemdetect('-z', dd.path)
```

Comment:
We cannot use pmemdetect here. Pmemdetect uses util_file_get_size->util_file_fd->pmem2_config_geT_file_size to read size. It means you tests funcX using funcX. Please use ndctl cli